add document for the configmap fields (#252)
openshift/managed-upgrade-operator: docs/configmap.md

# Config Map

## About
The `configmap` which used to tune the `managed-upgrade-operator`. It has various configurable values.

## How to use it

### For OSD:
The `configmap` will be synced to the clusters from the hive cluster via selectorsyncset. You can update the `configmap` via [`managed-cluster-config`](https://github.com/openshift/managed-cluster-config/tree/master/deploy/managed-upgrade-operator-config).

### For ARO:

> TDB

Reviewer: My personal preference is to leave out deployment-specific stuff in these docs and keep it generic (ie. "how to use it" from the perspective of a non-Red Hatter who won't know what Hive is), and maintain that in Source (as in, "The Source") documentation. But happy to hear what the rest of the team want to do here and go with the majority feeling.

Author: how about just mention the managed cluster config is the place for OSD? but not mention hive and syncset. 
for people easy to reference/locating the place where to update

## Configurable knobs
#### configManager

Please refer to the doc [`configmanager`](../configmanager.md)

#### maintenance

| Key | Description |
| --- | --- |
| controlPlaneTime | maintenance window created in alertmanager for controlplane upgrade in minutes including all the low and medium alerts, default is 90 |

> | ignoredAlerts | a list of particular alerts to silent which is not covered by the default maintenance |

Reviewer: ```suggestion
| ignoredAlerts | a list of particular alerts to silence which are not covered by the default maintenance |
```

| controlPlaneCriticals | critial alerts might fire during controlplane upgrade |

Example:
```
    maintenance:
      controlPlaneTime: 90
      ignoredAlerts:
        controlPlaneCriticals:
        - ClusterOperatorDown
        - ClusterOperatorDegraded
```

#### scale

| Key | Description |
| --- | --- |
| timeOut | timeout window for the extra workload scale up in minutes, default is 30 |

Example:
```
    scale:
      timeOut: 30
```

#### upgradeWindow

| Key | Description |
| --- | --- |

> | delayTrigger | a time window to indicate the the cluster upgrade is deleyed against the schedule in minutes, default is 30 |

Reviewer: ```suggestion
| delayTrigger | a time window to indicate that the cluster upgrade is delayed against the schedule in minutes, default is 30 |
```

> | timeOut | a time window to indicate the whole upgrade process should be finished within in minutes, default is 120 |

Reviewer: ```suggestion
| timeOut | a time window within which the upgrade process should have started before it is considered as "failed". Measured in minutes, default is 120 |
```

Example:
```
    upgradeWindow:
      delayTrigger: 30
      timeOut: 120
```

#### nodeDrain

| Key | Description |
| --- | --- |

> | timeOut | a time window to trigger the force node drain stretegy |

Reviewer: ```suggestion
| timeOut | a time window to trigger the force node drain strategy, measured in minutes |
```

> | expectedNodeDrainTime | expected time in minutes for a single node drain to be finished, sed to setup the maintenance window |

Reviewer: ```suggestion
| expectedNodeDrainTime | expected time in minutes for a single node drain to be finished, used to setup the maintenance window |
```

Example:
```
    nodeDrain:
      timeOut: 45
      expectedNodeDrainTime: 8
```

#### healthCheck

| Key | Description |
| --- | --- |

> | ignoredCriticals | a list of critical alerts need to be ignored in health check to unblock the upgrade process |

Reviewer: ```suggestion
| ignoredCriticals | a list of critical alerts which need to be ignored in the health check to unblock the upgrade process |
```

> | ignoredNamespaces | a list of namespaces need to be ignored in health check to unblock the upgrade process |

Reviewer: ```suggestion
| ignoredNamespaces | a list of namespaces which need to be ignored in the health check to unblock the upgrade process |
```

Example:
```
    healthCheck:
      ignoredCriticals:
      - DNSErrors05MinSRE
      - FluentdNodeDown
      ignoredNamespaces:
      - openshift-logging
      - openshift-redhat-marketplace
```

#### extDependencyAvailabilityChecks

| Key | Description |
| --- | --- |

> | http | a list of external dependencies to check before the upgrade can be triggered |

Reviewer: ```suggestion
| http | a list of external HTTP(S) dependencies to check before the upgrade can be triggered. Dependencies must return a 200 HTTP Status Code to be considered healthy. |
```

Example:
```
    extDependencyAvailabilityChecks:
      http:
        timeout: 10
        urls:

>           - www.example.com

Reviewer: Not actually sure what happens in MUO if a URL is specified without a leading protocol. (something to test)

```suggestion
          - https://www.example.com
```

# Config Map

- [Config Map](#config-map)
  - [About](#about)
  - [How to use it](#how-to-use-it)
    - [For OSD](#for-osd)
    - [For ARO](#for-aro)
  - [Configurable parameters](#configurable-parameters)
    - [configManager](#configmanager)
    - [maintenance](#maintenance)
    - [scale](#scale)
    - [upgradeWindow](#upgradewindow)
    - [nodeDrain](#nodedrain)
    - [healthCheck](#healthcheck)
    - [extDependencyAvailabilityChecks](#extdependencyavailabilitychecks)

## About
The `configmap` which used to tune the `managed-upgrade-operator`. It has various configurable values.

## How to use it

### For OSD
You can update the `configmap` via [`managed-cluster-config`](https://github.com/openshift/managed-cluster-config/tree/master/deploy/managed-upgrade-operator-config).

### For ARO
TBD

## Configurable knobs
#### configManager

> Please refer to the doc [`configmanager`](../configmanager.md)

Reviewer 2: Change `../configmanager.md` to `./configmanager.md` since it's in the `docs` directory itself.

#### maintenance

| Key | Description |
| --- | --- |
| controlPlaneTime | maintenance window created in alertmanager for controlplane upgrade in minutes including all the low and medium alerts, default is 90 |
| ignoredAlerts | a list of particular alerts to silence which are not covered by the default maintenance |

> | controlPlaneCriticals | critial alerts might fire during controlplane upgrade |

Reviewer 2: s/critial/critical/

Example:
```
    maintenance:
      controlPlaneTime: 90
      ignoredAlerts:
        controlPlaneCriticals:
        - ClusterOperatorDown
        - ClusterOperatorDegraded
```

#### scale

| Key | Description |
| --- | --- |
| timeOut | timeout window for the extra workload scale up in minutes, default is 30 |

Example:
```
    scale:
      timeOut: 30
```

#### upgradeWindow

| Key | Description |
| --- | --- |

> | delayTrigger | a time window to indicate that the cluster upgrade is deleyed against the schedule in minutes, default is 30 |

Reviewer 2: s/deleyed/delayed/

Reviewer 3: ```suggestion
| delayTrigger | a time window to indicate that the cluster upgrade is delayed against the schedule in minutes, default is 30 |
```

| timeOut | a time window which the upgrade process should have started before it is considered as "failed". Measured in minutes, default is 120 |

Example:
```
    upgradeWindow:
      delayTrigger: 30
      timeOut: 120
```

#### nodeDrain

| Key | Description |
| --- | --- |

> | timeOut | a time window to trigger the force node drain stretegy, meatured in minutes |

Reviewer 2: s/stretegy/strategy/

Reviewer 3: ```suggestion
| timeOut | a time window to trigger the force node drain strategy, measured in minutes |
```

Reviewer 2: s/meatured/measured/
